backend/src/main/java/com/bakdata/conquery/sql/conversion/cqelement/CQConceptConverter.java

```diff
@@ -34,50 +32,178 @@ public Class<CQConcept> getConversionClass() {
+	}
+
+	/**
+	 * selects:
+	 * - all of previous steps
+	 * - transformed columns with selects
+	 */
+	private QueryStep buildEventSelectQueryStep(
+			ConversionContext context,
+			CQTable table,
+			String conceptLabel, QueryStep previous
+	) {
+		if (table.getSelects().isEmpty()) {
+			return previous;
+		}
+
+		ConceptSelects eventSelectSelects = this.prepareEventSelectSelects(context, table, previous);
+		String eventSelectCteName = "concept_%s_event_select".formatted(conceptLabel);
+
+		return QueryStep.builder()
+						.cteName(eventSelectCteName)
+						.fromTable(QueryStep.toTableLike(previous.getCteName()))
+						.selects(eventSelectSelects)
+						.conditions(List.of())
+						.predecessors(List.of(previous))
+						.build();
+	}
+
+	/**
+	 * selects:
+	 * - all of previous step
+	 * - remove filter
+	 */
+	private QueryStep buildEventFilterQueryStep(
+			ConversionContext context,
+			CQTable table,
+			String conceptLabel,
+			QueryStep previous
+	) {
+		if (table.getFilters().isEmpty()) {
+			return previous;
+		}
+
+		ConceptSelects eventFilterSelects = this.prepareEventFilterSelects(previous);
+		List<Condition> eventFilterConditions = this.buildEventFilterConditions(context, table);
+		String eventFilterCteName = "concept_%s_event_filter".formatted(conceptLabel);
+
+		return QueryStep.builder()
+						.cteName(eventFilterCteName)
+						.fromTable(QueryStep.toTableLike(previous.getCteName()))
+						.selects(eventFilterSelects)
+						.conditions(eventFilterConditions)
+						.predecessors(List.of(previous))
+						.build();
 	}
 
-	private String getPrimaryColumnName(CQTable table) {
-		// TODO add primary column to table
-		return "pid";
+	/**
+	 * selects:
+	 * - all of previous step
+	 */
+	private QueryStep buildFinalQueryStep(String conceptLabel, QueryStep previous) {
+		ConceptSelects finalSelects = ((ConceptSelects) previous.getQualifiedSelects());
+		return QueryStep.builder()
+						.cteName("concept_%s".formatted(conceptLabel))
+						.fromTable(QueryStep.toTableLike(previous.getCteName()))
+						.selects(finalSelects)
+						.conditions(List.of())
+						.predecessors(List.of(previous))
+						.build();
 	}
 
-	private List<Field<?>> getConceptSelects(CQTable table, ConversionContext context) {
+	private ConceptSelects prepareDateRestrictionSelects(CQConcept node, QueryStep previous) {
+		ConceptSelects.ConceptSelectsBuilder selectsBuilder = ((ConceptSelects) previous.getQualifiedSelects()).toBuilder();
+		selectsBuilder.dateRestriction(Optional.empty());
+		if (node.isExcludeFromTimeAggregation()) {
+			selectsBuilder.validityDate(Optional.empty());
+		}
+		return selectsBuilder.build();
+	}
+
+	private List<Condition> buildDateRestriction(ConversionContext context, QueryStep previous) {
+		return ((ConceptSelects) previous.getSelects()).getDateRestriction()
+													   .map(dateRestrictionColumn -> getDateRestrictionAsCondition(context, previous, dateRestrictionColumn))
+													   .orElseGet(List::of);
```

Comment:
sollte es nicht eher Collections::emptyList sein?

Reply:
Warum? `List.of()` gibt eine statische immutable List zurück und wir brauchen hier keine mutability